DataFormats/L1TParticleFlow/src/HPSPFTau.cc

```diff
@@ -0,0 +1,87 @@
+#include "DataFormats/L1TParticleFlow/interface/HPSPFTau.h"
+
+// default constructor
+l1t::HPSPFTau::HPSPFTau()
+    : tauType_(kUndefined),
+      sumChargedIso_(0.),
+      sumNeutralIso_(0.),
+      sumCombinedIso_(0.),
+      sumChargedIsoPileup_(0.),
+      rhoCorr_(0.),
+      passTightIso_(false),
+      passMediumIso_(false),
+      passLooseIso_(false),
+      passVLooseIso_(false),
+      passTightRelIso_(false),
+      passMediumRelIso_(false),
+      passLooseRelIso_(false),
+      passVLooseRelIso_(false) {}
+
+// destructor
+l1t::HPSPFTau::~HPSPFTau() {}
+
+// print to stream
+ostream& operator<<(ostream& os, const l1t::HPSPFTau& l1PFTau) {
+  os << "pT = " << l1PFTau.pt() << ", eta = " << l1PFTau.eta() << ", phi = " << l1PFTau.phi()
+     << " (type = " << l1PFTau.tauType() << ")" << std::endl;
+  os << "lead. ChargedPFCand:" << std::endl;
+  if (l1PFTau.leadChargedPFCand().isNonnull()) {
+    printPFCand(os, *l1PFTau.leadChargedPFCand(), l1PFTau.primaryVertex());
+  } else {
+    os << " N/A" << std::endl;
+  }
+  os << "seed:";
+  if (l1PFTau.isChargedPFCandSeeded()) {
+    os << " chargedPFCand";
+  } else if (l1PFTau.isPFJetSeeded()) {
+    os << " PFJet";
+  } else
+    assert(0);
```

Comment:
a clearer error message (exception) could be provided here

Reply:
It will now throw a cms::Exception with more information.